types/shares.go

```diff
@@ -174,3 +168,15 @@ func zeroPadIfNecessary(share []byte, width int) []byte {
 	}
 	return share
 }
+
+// appendCopy makes a copy of slice a and appends slices b to that copy
+func appendCopy(a []byte, b ...[]byte) []byte {
+	c := make([]byte, len(a))
+	copy(c, a)
+
+	for _, bi := range b {
+		c = append(c, bi...)
```

Comment:
This will append to a slice with capacity `len(a)` fully used up as we've already copied `a`. To make room for bi, `c`'s capacity will be doubled for each time the append doesn't have enough room in `c`. IMO, we should make use of the fact that we know the size of the resulting slice upfront instead (it's len(a)+ length of each element passed in `b`). I'd still prefer the explicit appends in place (as in the comments) but if you really want to make it even more readable, sum up the `len(bi)`, create `c` with make([]byte, len(a) + <that sum>) and then append each thing in there (`a` and the `bi`s).

Reply:
that makes sense, started using appends here https://github.com/lazyledger/lazyledger-core/pull/235/commits/8bba9d9622ba0b791360f7bf633e1e4409f0127a